### ISSUE TITLE
[BUG REPORT] - "Too many requests"

### ISSUE THREAD

**Issue by joe024132** [SegoCode/4cget#4]
### Description

When downloading any thread, most of the files will be downloaded as raw text files containing the text: "Too Many Requests"
![image](https://github.com/user-attachments/assets/d14f1216-67a2-4b96-a1a1-d629b5ed4688)
![image](https://github.com/user-attachments/assets/4a07bf53-163d-4774-a8ad-40a69a90c2e3)

### Steps to reproduce

I used https://boards.4chan.org/wsg/thread/5731328 as a test.

### Expected behavior

All of the images should correctly download.

### Actual behavior

Only some images correctly download, and the rest are just raw text files.

### OS version

Windows 11

### Confirmation

- [X] I performed a [search of the issue tracker](https://github.com/SegoCode/4cget/issues) to avoid opening a duplicate issue
- [X] I understand that not filling out this template correctly may lead to the issue being closed

**Comment by SegoCode** [comment]
Hi there, 

Well that happens because you have reached a certain limit from your IP, you can connect to a VPN and continue downloading from there or wait a while. Some solutions would be to implement some “interval” between downloads or to be able to define proxies. Any other ideas?

**Comment by M0ller** [comment]
I got this to and added a interval that paused for a few seconds and then tried again for a couple of times. It works better. But also adding a millisecond wait before each go routine adds a bit more help as well. 

Also did a check to only download the file if it is not already downloaded and if the request succeeds.

Could add this as a param to the command if it should apply the wait's.

**Comment by SegoCode** [comment]
> I got this to and added a interval that paused for a few seconds and then tried again for a couple of times. It works better. But also adding a millisecond wait before each go routine adds a bit more help as well.
> 
> Also did a check to only download the file if it is not already downloaded and if the request succeeds.
> 
> Could add this as a param to the command if it should apply the wait's.

Yeah, definitely dont download the text “too many request” has to be implemented. And some proxy and interval parameters

**Comment by M0ller** [comment]
Yeah true. Maybe a config json file that have some default values for wait timer etc and then later a proxy  config or such could be added in it if / when it is developed. That maybe is the way to go?

**Comment by SegoCode** [comment]
I made this snipped, in theory, if the flags are at the beginning they can be parsed until they reach the string URL

```go
	// Manually parse flags and positional arguments
	var args []string
	for i := 1; i < len(os.Args); i++ {
		arg := os.Args[i]
		if arg == "--" {
			// All remaining args are positional
			args = append(args, os.Args[i+1:]...)
			break
		}
		if strings.HasPrefix(arg, "-") {
			// Flag
			fs.Parse(os.Args[i:])
			break
		}
		// Positional argument
		args = append(args, arg)
	}

	// After parsing flags, any remaining arguments are positional
	args = append(args, fs.Args()...)

```

I made a branch https://github.com/SegoCode/4cget/tree/develop-flags to test and keep code this implementation. https://github.com/SegoCode/4cget/commit/888dc2324e74a07cd2336a8c5b252e5098471f29

**Comment by SegoCode** [comment]
Restarting development on this 🎉

4cget now reads the status first and skips 403/429 instead of calling `io.Copy`. On those statuses it prints a Cloudflare hint: low IP trust score, browser challenge required. 429 also mentions `--sleep`.

Cloudflare scores each GET on `boards.4chan.org` from IP, ASN, TLS fingerprint, and User-Agent. Normal browser traffic from the same IP raises that score. Open the thread in a browser, browse a bit, then retry the CLI.

`--sleep N` serializes downloads with N seconds between files. That is the interval we talked about here. `--monitor` already waits between thread polls, so the two flags cannot run together.

I dropped proxy support. 429 on `i.4cdn.org` is a rate limit on that IP. Spacing requests is the control we have without another hop.

If 429 shows up again, raise `--sleep` and retry.
